cpp-lite/include/mxnet-lite/ndarray.h

```diff
@@ -0,0 +1,284 @@
+/*
```

Comment:
We should use the pimpl idiom for the public headers such as ndarray.

Reply:
this is intended to be header only

Reply:
Header only is usually good when you have templated classes and they get instantiated at the place of usage. What is the advantage in this case?